src/cache.rs

```diff
@@ -144,20 +170,39 @@ impl Cache {
 
     /// Check for pages for a given platform in one of the given languages.
     fn find_page_for_platform(
-        name: &str,
+        page_name: &str,
         cache_dir: &Path,
         platform: &str,
         language_dirs: &[String],
-    ) -> Option<PathBuf> {
+    ) -> Option<PageLookupResult> {
         language_dirs
             .iter()
-            .map(|lang_dir| cache_dir.join(lang_dir).join(platform).join(name))
+            .map(|lang_dir| cache_dir.join(lang_dir).join(platform).join(page_name))
             .find(|path| path.exists() && path.is_file())
+            .map(|file| PageLookupResult::with_page(file))
+    }
+
+    /// Look up custom patch (<name>.patch). If it exists, store it in a variable.
+    fn find_patch(patch_name: &str, maybe_custom_dir: Option<&Path>) -> Option<PathBuf> {
+        if let Some(custom_dir) = maybe_custom_dir {
+            let patch_path = custom_dir.join(patch_name);
+            if patch_path.exists() && patch_path.is_file() {
+                return Some(patch_path);
+            }
+        }
+        None
     }
 
     /// Search for a page and return the path to it.
-    pub fn find_page(&self, name: &str, languages: &[String]) -> Option<PathBuf> {
+    pub fn find_page(
+        &self,
+        name: &str,
+        languages: &[String],
+        config: &Config,
```

Comment:
This function doesn't need the full config, just the `custom_pages_dir` field, so in my opinion it should only get passed that to be more specific.